### PR TITLE
Buy Vehicle Part 2

### PR DESCRIPTION
**Pull request by pmoons**
- [x] Add OwnerBuyVehicleScene
- [x] Create query that displays all vehicles that do not have an associated owner
- [x] Allow user to choose new car
- [ ] Associate user with that vehicle

### CODE REVIEW

File: src/Tesla/Model/Vehicle.java [hunk 1]
```diff
@@ -19,11 +22,30 @@
 	public int Mileage;
 	public String Color;
 	
+	public String getVin() { return this.Vin; }
```
Comment (pmoons) [inline src/Tesla/Model/Vehicle.java]: Key thing to note here: We don't need a million accessor methods returning a `StringProperty`.  The `TableColumn`'s `setCellValueFactory` just looks for a getter with the method signature `get[AttributeName]`.

**Comment by natesuver** [comment]
For table column that might be true, but is that true for elements like
textboxes that do 2 way binding?  Vehicle class will need to support both
table view binding and data entry widget binding

**Comment by pmoons** [comment]
@natesuver It probably won't do two-way data binding. Is there an instance in the app so far where updating the model (such as clicking submit when updating a vehicle's or owner's information) immediately jumps the user to a new view that contains the model's updated information?  I thought it usually just jumps back to the main menu screen.

If it did jump back to the form and was displaying the newly updated information, then I can see a case for two-way data binding.  If the app doesn't do that though, then why have two way data binding?  Re-accessing the menu to see the user's information is just going to initiate another fetch to the server.  We'd only really be making use of "one-way" data binding (ie fetch results -> text boxes).

If we're implementing the `Owner` class the way we are to map the retrieved attributes to the text box fields "automagically" in the name of convenience, I'd have to disagree.  All those attribute accessor methods and overhead make the model extremely noisy and difficult to follow.

**Comment by natesuver** [comment]
Yeah, binding to the textboxes on the vehicle add screen requires bi
directional (2 way) binding.  That's how the model and the textbox keep
each other in sync.

**Pull request merged**